ui/components/app/toast-master/toast-master.js

```diff
@@ -0,0 +1,299 @@
+
+  const [hideConnectAccountToast, setHideConnectAccountToast] = useState(false);
+  const account = useSelector(getSelectedAccount);
+
+  // If the account has changed, allow the connect account toast again
+  const prevAccountAddress = usePrevious(account?.address);
+  if (account?.address !== prevAccountAddress && hideConnectAccountToast) {
+    setHideConnectAccountToast(false);
+  }
+
+  const showConnectAccountToast = useSelector((state) =>
+    selectShowConnectAccountToast(state, account),
+  );
+
+  const activeTabOrigin = useSelector(getOriginOfCurrentTab);
+
+  return (
+    Boolean(!hideConnectAccountToast && showConnectAccountToast) && (
+      <Toast
+        dataTestId="connect-account-toast"
+        key="connect-account-toast"
+        startAdornment={
+          <AvatarAccount
+            address={account.address}
+            size={AvatarAccountSize.Md}
+            borderColor={BorderColor.transparent}
+          />
+        }
+        text={t('accountIsntConnectedToastText', [
+          account?.metadata?.name,
+          getURLHost(activeTabOrigin),
+        ])}
+        actionText={t('connectAccount')}
+        onActionClick={() => {
+          // Connect this account
+          dispatch(addPermittedAccount(activeTabOrigin, account.address));
+          // Use setTimeout to prevent React re-render from
+          // hiding the tooltip
+          setTimeout(() => {
+            // Trigger a mouseenter on the header's connection icon
+            // to display the informative connection tooltip
+            document
+              .querySelector(
+                '[data-testid="connection-menu"] [data-tooltipped]',
+              )
+              ?.dispatchEvent(new CustomEvent('mouseenter', {}));
+          }, 250 * MILLISECOND);
+        }}
+        onClose={() => setHideConnectAccountToast(true)}
+      />
+    )
+  );
+}
+
+function SurveyToastMayDelete() {
+  const t = useI18nContext();
+
+  const showSurveyToast = useSelector(selectShowSurveyToast);
+
+  return (
+    showSurveyToast && (
+      <Toast
+        key="survey-toast"
+        startAdornment={
+          <Icon name={IconName.Heart} color={IconColor.errorDefault} />
+        }
+        text={t('surveyTitle')}
+        actionText={t('surveyConversion')}
+        onActionClick={() => {
+          global.platform.openTab({
+            url: SURVEY_LINK,
+          });
+          setSurveyLinkLastClickedOrClosed(Date.now());
+        }}
+        onClose={() => {
+          setSurveyLinkLastClickedOrClosed(Date.now());
+        }}
+      />
+    )
+  );
+}
+
+function PrivacyPolicyToast() {
+  const t = useI18nContext();
+
+  const { showPrivacyPolicyToast, newPrivacyPolicyToastShownDate } =
+    useSelector(selectShowPrivacyPolicyToast);
+
+  // If the privacy policy toast is shown, and there is no date set, set it
+  if (showPrivacyPolicyToast && !newPrivacyPolicyToastShownDate) {
+    setNewPrivacyPolicyToastShownDate(Date.now());
```

Comment:
This might be ran several times, perhaps would be cleaner to move into a useEffect